Migrate to yarn

jroebu14 opened the pull request:
Resolves https://github.com/bbc/psammead/issues/4331

**Overall change:** 
Replaces npm with yarn for package management.

**Code changes:**

NPM to Yarn command translations:
- `npm install` -> `yarn install` install packages
- `npm ci` -> `yarn install --frozen-lockfile` for deterministic builds based on lockfile
- `npm install --quiet` -> `yarn install --silent` installs without console noise
-  `npm update` -> `yarn upgrade` upgrades all packages
- `npm install some-package` -> `yarn add some-package` installs some-package
- `npm install --no-package-lock` -> `yarn install --no-lockfile` do no generate a lock file on installing
---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

Commit:
create yarn.lock using yarn import

Commit:
update enforceVersions to use yarn

Commit:
update package.json to use yarn commands

Commit:
update github actions to use yarn

Commit:
update Talos to use yarn

Commit:
update yeoman to use yarn

Commit:
update docs to use yarn

Commit:
yarn import to generate yarn.lock and rm package-lock.json

Commit:
update lerna config

Commit:
prevent npm installs. add doc on installing and using yarn

Commit:
update shell scripts to use yarn

Commit:
update talos to use yarn

Commit:
update readme.md

Commit:
replace .npmrc with .yarnrc

Commit:
use correct yarn flag --no-lockfile in lerna.json

andrewscfc commented on scripts/check-package-manager.js:
```diff
@@ -0,0 +1,19 @@
+const ensureYarn = () => {
+  // If npm is being run, print a message and cause 'npm install' to fail.
+  if (!process.env.npm_execpath.includes('yarn')) {
+    console.log(
```
Maybe add a lint ignore here to suppress this warning?

andrewscfc commented on scripts/enforceVersions.sh:
```diff
@@ -12,15 +12,15 @@ if [[ $(pwd) =~ .+psammead ]]; then
+  if [[ $(yarn -v) != $desired_yarn_version ]]; then
     tput setaf 1
-    echo "\nOh no! You're not using our version of npm, please move to version $desired_npm_version in order to reduce bundling inconsistency"
+    echo "\nOh no! You're not using our version of yarn, please move to version $desired_yarn_version in order to reduce bundling inconsistency"
```
Did you do this on the Simorgh branch, I don't remember seeing it, the same script is in that repo too?

jroebu14 replied:
Yep this was in the original Simorgh PR

andrewscfc reviewed (commented):
Looks good, few thoughts:
- Did you you use [`yarn import`(https://classic.yarnpkg.com/en/docs/cli/import/) for the top level package lock? We needed to do this for Simorgh on the call the other day, it may well be necessary for psammead too? I guess the others were regenerated by the existing tooling.
- It seems you've wisely chosen to not remove lerna in this PR, I guess we may have a follow-up PR to either
  - Remove lerna and replicate it's functionality using yarn's built in features
  - Or investigate our existing issues with lerna and try and get local package linking working for example. A new version of lerna has appeared which gives us a bit more confidence in sticking with lerna: https://github.com/lerna/lerna/issues/2703#issuecomment-777089520

Commit:
fix preinstall script

jroebu14 commented:
@andrewscfc yeh I used `yarn import && rm package-lock.json` in the top level and then `lerna exec -- yarn import && rm package-lock.json` to do the same in every package.

Yeh this PR is just to use Yarn for installing dependencies and switching to Yarn commands. Anything else will be separate PRs.

Commit:
ignore console.log warning

chris-hinds commented on .github/workflows/psammead-publish.yml:
```diff
@@ -22,10 +22,10 @@ jobs:
 
       - name: Install Node Modules
-        run: npm run ci:packages
+        run: yarn run ci:packages
```
It's small and doesn't make much different but I think with yarn you can also remove the `run` command so just `yarn ci:packages` should work.

Not crucial to this work though.

jroebu14 replied:
Removed `run` from all yarn commands in [this commit](https://github.com/bbc/psammead/pull/4335/commits/a335bd91d50a5808b9a685b3080385b4b6f5ffb1)

chris-hinds commented on .github/workflows/psammead-change-scanner.yml:
```diff
@@ -17,7 +17,7 @@ jobs:
 
       - name: Install Node Modules
-        run: npm run ci:packages
+        run: yarn run ci:packages
```
Looks like the SetupNode GH action comes preinstalled with yarn! Nice 👍

Commit:
remove `run` from yarn commands

Commit:
fix package bumping script

Commit:
Merge branch 'latest' into migrate-to-yarn

Commit:
Merge branch 'latest' into migrate-to-yarn

Commit:
Bump package versions

Bump package versions for psammead, psammead-amp-geo, psammead-brand, psammead-bulleted-list, psammead-bulletin, psammead-byline, psammead-caption, psammead-consent-banner, psammead-content-anchor, psammead-copyright, psammead-embed-error, psammead-episode-list, psammead-figure, psammead-grid, psammead-heading-index, psammead-headings, psammead-image-placeholder, psammead-image, psammead-inline-link, psammead-live-label, psammead-media-indicator, psammead-media-player, psammead-most-read, psammead-navigation, psammead-oembed, psammead-paragraph, psammead-play-button, psammead-podcast-promo, psammead-radio-schedule, psammead-script-link, psammead-section-label, psammead-sitewide-links, psammead-social-embed, psammead-story-promo-list, psammead-story-promo, psammead-timestamp, psammead-useful-links, psammead-visually-hidden-text, psammead-timestamp-container, gel-foundations, moment-timezone-include, psammead-assets, psammead-calendars, psammead-detokeniser, psammead-locales, psammead-rich-text-transforms, psammead-storybook-helpers, psammead-styles, psammead-test-helpers, web-vitals

Commit:
Update changelogs

Updates changelogs for psammead, psammead-amp-geo, psammead-brand, psammead-bulleted-list, psammead-bulletin, psammead-byline, psammead-caption, psammead-consent-banner, psammead-content-anchor, psammead-copyright, psammead-embed-error, psammead-episode-list, psammead-figure, psammead-grid, psammead-heading-index, psammead-headings, psammead-image-placeholder, psammead-image, psammead-inline-link, psammead-live-label, psammead-media-indicator, psammead-media-player, psammead-most-read, psammead-navigation, psammead-oembed, psammead-paragraph, psammead-play-button, psammead-podcast-promo, psammead-radio-schedule, psammead-script-link, psammead-section-label, psammead-sitewide-links, psammead-social-embed, psammead-story-promo-list, psammead-story-promo, psammead-timestamp, psammead-useful-links, psammead-visually-hidden-text, psammead-timestamp-container, gel-foundations, moment-timezone-include, psammead-assets, psammead-calendars, psammead-detokeniser, psammead-locales, psammead-rich-text-transforms, psammead-storybook-helpers, psammead-styles, psammead-test-helpers, web-vitals

Commit:
use yarn in storybook preview action

Commit:
add yarn ci script to remove invalid transitive dependencies before new install

Commit:
merge latest

Commit:
Merge branch 'latest' into migrate-to-yarn

Commit:
fix ci:packages script which uses lerne exec

Commit:
fix ci:packages not installing packages

Commit:
fix bumpPackages unit test

Commit:
install all packages in storybook-preview GH action
